sklearn/naive_bayes.py

```diff
@@ -51,8 +51,10 @@ def _joint_log_likelihood(self, X):
         I.e. ``log P(c) + log P(x|c)`` for all rows x of X, as an array-like of
         shape (n_samples, n_classes).
 
-        predict, predict_proba, and predict_log_proba pass the input through
-        _check_X and handle it over to _joint_log_likelihood.
+        Public methods predict, predict_proba, predict_log_proba, and
+        predict_joint_log_proba pass the input through _check_X before handing it
+        over to _joint_log_likelihood. The term "joint log likelihood" is used
```

Comment:
I would do put in public documentation pointer toward private API.
Rather than referring to _check_X, you could say input validation.
Idem for _joint_log_likelihood

Reply:
An alternative is to simply move this text as comments in the code
eg within the class code block, and not have in the top docstring.
Users do not really need to know internals anyway.

Reply:
I am not sure I am following you. This is a private method referring to public methods, no the other way around. This docstring never makes it to the public documentation. It is needed to clarify the purpose and the scope of the abstract method to developers.

Reply:
Arff indeed. I thought it was a public docstring.